pandas/core/arrays/period.py

```diff
@@ -536,11 +537,14 @@ def _sub_period(self, other):
     @Appender(dtl.DatetimeLikeArrayMixin._addsub_int_array.__doc__)
     def _addsub_int_array(
             self,
-            other,   # type: Union[Index, ExtensionArray, np.ndarray[int]]
-            op      # type: Callable[Any, Any]
+            other,      # type: Union[ExtensionArray, np.ndarray[int]]
+            op          # type: Callable[Any, Any]
     ):
         # type: (...) -> PeriodArray
 
+        # TODO: ABCIndexClass is a valid type for other but had to be excluded
```

Comment:
Comment explains it, but just to clarify this PR doesn't necessarily check that comments are correct as much as gets them to pass linting. I figure it makes sense to validate the comments themselves when moving over to Py3 syntax, so this was the quick way of getting this to pass